dask/array/ghost.py

```diff
@@ -245,9 +250,11 @@ def nearest(x, axis, depth):
     return concatenate([l, x, r], axis=axis)
 
 
-
 def constant(x, axis, depth, value):
     """ Add constant slice to either side of array """
+    if depth == 0:
+        return x
```

Comment:
This seems to be a common fix.  Should this be handled where `constant` and the others are called?